### PR TITLE
Add Mock form

### PR DESCRIPTION
**Pull request by Mottie**
## Description

Add a mock form to [implement tutorial and documentation pages](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/forms/), and to show alternative patterns and best practices. This should help with onboarding engineers, and be a showcase for implementation of newer patterns and workarounds for issues.

- This form will _never_ be available in production.
- The plan is for this is to be a living, constantly evolving form that gets updated as best practices change

cc: @micahchiang

## Original issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/39437

## Testing done

None (yet?)

## Screenshots

<details><summary>Checkbox group w/validation page</summary>

checkbox group pattern and va-checkbox-group widget examples</details>

## Acceptance criteria
- [x] Add mock form for showcasing the form system

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by pdavies88** [comment]
@Mottie so back in #19901 I had created a Mock Form named `mock-sip-from` do we want to remove that? It seems like this PR updates some of the things I did with that ie: `FORM_XX_123` which is what I had randomly named it.

The main purpose of creating that form was so that we could test:
- src/platform/forms/tests/save-in-progress/01-sip-autosave.cypress.spec.js
- src/platform/forms/tests/save-in-progress/01-sip-finish-later.cypress.spec.js
- src/platform/forms/tests/save-in-progress/01-sip-load-fail.cypress.spec.js
- src/platform/forms/tests/save-in-progress/01-sip-review.cypress.spec.js
Without having to use the Health Care Application and I think having two mock forms might not make sense.

**Comment by Mottie** [comment]
Hi @pdavies88!

I didn't include any testing in this form. I meant for it to be more of a showcase for [all the tutorial examples from the docs](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/forms), and also include newer patterns. I'll leave removing the `mock-sip-form` path up to you. This form serves a different purpose.

**Comment by pdavies88** [comment]
> Hi @pdavies88!
> 
> I didn't include any testing in this form. I meant for it to be more of a showcase for [all the tutorial examples from the docs](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/forms), and also include newer patterns. I'll leave removing the `mock-sip-form` path up to you. This form serves a different purpose.

@Mottie cool sounds great I am going to leave it in then since reworking to fit in with the new form is something that should probably be ticketed and prioritized. Saw you added back in `FORM_XX_123` was going to say removing it might cause some build issues. 👍

**Review by pdavies88 — APPROVED** [review]

**Review by micahchiang — APPROVED** [review]
This is awesome. Thanks so much for putting the time in to creating this @Mottie.

File: src/applications/_mock-form/components/GroupCheckboxWidget.jsx
Comment (bkjohnson) [inline src/applications/_mock-form/components/GroupCheckboxWidget.jsx:4]: I know this is how the react bindings are imported in some places, but can we use the true path instead of [the aliased one](https://github.com/department-of-veterans-affairs/vets-website/blob/ac60e8f296360efbc597516f5947139b25b4210f/.babelrc#L63)? I'd like to eventually get rid of that.

There's also an [example in the documentation](https://design.va.gov/documentation/developers#react-applications):

> ```jsx
> import { VaExampleComponent } from "@department-of-veterans-affairs/component-library/dist/react-bindings";
> ```

**Review by bjmccotter7192 — COMMENTED** [review]
LGTM! Awesome job this mock form is sweet!

File: src/applications/_mock-form/pages/directDeposit.js
Comment (bkjohnson) [inline src/applications/_mock-form/pages/directDeposit.js:55]: Do we need this file/schema? Could the example use the `directDeposit` definition that the forms library already has?

https://github.com/department-of-veterans-affairs/vets-website/blob/95a9fd72a66b4abddd788179f9b5fb684c122aff/src/platform/forms-system/src/js/definitions/directDeposit.jsx

File: src/applications/_mock-form/containers/IntroductionPage.jsx
Comment (bkjohnson) [inline src/applications/_mock-form/containers/IntroductionPage.jsx:72]: We're trying to discourage this, so could this use `<va-process-list>` as an example of a best practice?

File: src/applications/_mock-form/helpers.js
Comment (bkjohnson) [inline src/applications/_mock-form/helpers.js:7]: This seems like it might be misleading or I'm just confused. The class id defined in `src/applications/pensions/sass/pensions.scss`, but it doesn't look like _that_ sass file is imported into the sass for this mock form app.

https://github.com/department-of-veterans-affairs/vets-website/blob/7a4797f81610fce68477f46f36e31b2f281da655/src/applications/pensions/sass/pensions.scss#L38-L40

Also, if we do need that styling it looks like a utility class would work.

**Review by bkjohnson — COMMENTED** [review]
My comments are mostly about best practice stuff and patterns we want to encourage/discourage since this will be used as an example during onboarding.

File: src/applications/_mock-form/pages/contactInformation.js
Comment (Mottie) [reply src/applications/_mock-form/pages/contactInformation.js:8]: Hmm, I think we need to update the form builder template... I didn't change most of the patterns from it, including the Introduction page and included pages from the complex template.

File: src/applications/_mock-form/containers/ConfirmationPage.jsx
Comment (timwright12) [inline src/applications/_mock-form/containers/ConfirmationPage.jsx:2]: Can we use [date-fns](https://date-fns.org/) here instead of moment? We're trying to gradually get it out of the system because it's massive and deprecated

File: src/applications/_mock-form/helpers.js
Comment (timwright12) [inline src/applications/_mock-form/helpers.js:16]:
```suggestion
    aria-label="usdirectexpress.com, opens in new window"
    >
```

File: src/applications/_mock-form/containers/IntroductionPage.jsx
Comment (Mottie) [reply src/applications/_mock-form/containers/IntroductionPage.jsx:82]: I ended up unwrapping the `OMBInfo` component and adding an empty `<p />` tag before it.

File: src/applications/_mock-form/helpers.js
Comment (Mottie) [reply src/applications/_mock-form/helpers.js:7]: When I switched to using the `directDeposit` definition, this file ended up being removed.

File: src/applications/_mock-form/helpers.js
Comment (Mottie) [reply src/applications/_mock-form/helpers.js:16]: I ended up removing the `helpers.js` file since it was for the custom direct deposit component. I switched to using the platform definition for this - we'll need to eventually update the complex form template

**Comment by Mottie** [comment]
I forgot to ask, who should be the code owner for this?

**Comment by k80bowman** [comment]
@Mottie it should probably belong to `@department-of-veterans-affairs/vsp-design-system-fe`. Don't you think?

**Comment by Mottie** [comment]
@bkjohnson Is there anything else needed for this PR? I plan to add a few more example pages once this is merged.

**Review by bkjohnson — APPROVED** [review]
Nope, it looks fine to me :+1:

**Review by k80bowman — APPROVED** [review]

**Review by jhonny218 — APPROVED** [review]

**Pull request merged**
